# Perkenalan

Nama : Nailina Farah

Batch : FTDS_RMT_28

Dataset : rideshare_kaggle

# Model Inference

In [86]:
# Import Libaries
import pickle
import json
import pandas as pd
import numpy as np

# Load Model
> Load model yang telah dibuat pada notebook berbeda

In [87]:
# Load all models

with open('model_lin.pkl', 'rb') as  file_1: 
  model_lin = pickle.load(file_1)

with open('model_scaler.pkl', 'rb') as  file_2: 
  model_scaler = pickle.load(file_2)

with open('model_encoder.pkl', 'rb') as  file_3: 
  model_encoder = pickle.load(file_3)

with open('num_col.txt', 'r') as file_4:
  num_col = json.load(file_4)

with open('cat_col.txt', 'r') as file_5:
  cat_col = json.load(file_5)

# Create Data Inference
> Membuat data baru untuk mencoba model yang telah dibuat

Menginput value dari kolom-kolom pada data_inf

In [88]:
# create new data
data_inf = {
    'hour': 16,
    'day': 28,
    'month': 11,
    'cab_type': 'Uber',
    'name': 'UberX',
    'distance': 1.06,
    'short_summary': ' Mostly Cloudy ',
    'uvIndex': 1
}

data_inf = pd.DataFrame([data_inf])
data_inf

hour  day  month cab_type   name  distance    short_summary  uvIndex
0    16   28     11     Uber  UberX      1.06   Mostly Cloudy         1

Split kolom numerik dan kategorik

In [89]:
# Split between numerical and categorical columns
data_inf_num = data_inf[num_col]
data_inf_cat = data_inf[cat_col]
data_inf_cat

cab_type    short_summary   name
0     Uber   Mostly Cloudy   UberX

In [90]:
data_inf_num

hour  day  month  distance  uvIndex
0    16   28     11      1.06        1

Melakukan scaling pada kolom numerik dan encoding pada kolom kategorik. Karena encodingnya merupakan OneHotEncoder maka ditambah .toarray() agar memiliki dimensi yang sama.

In [91]:
# Feature scaling and encoding
data_inf_num_scaled = model_scaler.transform(data_inf_num)
data_inf_cat_encoded = model_encoder.transform(data_inf_cat).toarray()

# Concat data inference scaled and encoded
data_inf_final = np.concatenate([data_inf_num_scaled, data_inf_cat_encoded],axis=1)

Membuat predict untuk melihat hasil (target) dari features yang telah diinput dan diolah

In [92]:
# Predict new data inference using Linear Regression
y_pred_inf = model_lin.predict(data_inf_final)
print("Hasil prediksi harga perjalanan platform ride-hailing adalah $",round(y_pred_inf[0],2))

Hasil prediksi harga perjalanan platform ride-hailing adalah $ 6.6


Berdasarkan model regresi linear yang telah dibuat, hasil prediksi harga perjalanan platform ride-hailing dengan kondisi di jam 16, tanggal 28, bulan 11, tipe taksi Uber, tipe layanan UberX, jarak 1.06 KM, cuaca sebagian besar berawan, dan uvIndex 1 didapatkan harga sebesar $ 6.6.